Add sample for custom routing with Kong Gateway

sisp opened the pull request:
In this PR, I'm adding a sample for custom routing with Kong similar to https://knative.dev/docs/serving/samples/knative-routing-go/. Since it's not entirely obvious in my opinion how to exactly configure Kong for custom routing like path-based routing, I think would be a nice addition to the docs, hopefully saving others the time of figuring it out.

Let me know what you think.

Would you also like to have the draw.io source file of the diagram included in the images` folder?

abrennan89 commented on docs/serving/samples/kong-routing-go/README.md:
```diff
@@ -0,0 +1,273 @@
+---
+title: "Routing across multiple Knative services with Kong"
+linkTitle: "Routing services with Kong"
+weight: 1
+type: "docs"
+---
+
```
We don't include this frontmatter anymore, so it can be removed

abrennan89 commented on docs/serving/samples/kong-routing-go/README.md:
```diff
@@ -0,0 +1,273 @@
+type: "docs"
+---
+
+# Routing across multiple Knative services with Kong
```
```suggestion
# Routing across multiple Knative Services with Kong
```

abrennan89 commented on docs/serving/samples/kong-routing-go/README.md:
```diff
@@ -0,0 +1,273 @@
+type: "docs"
+---
+
+# Routing across multiple Knative services with Kong
+
+This example shows how to map multiple Knative services to different paths under
+a single domain name using [Kong Gateway](https://docs.konghq.com/gateway-oss/).
+Kong Gateway is a general-purpose reverse proxy and API gateway built on
+[Nginx](https://www.nginx.com/), therefore these directions can also be used to
+configure routing based on other request data such as headers, or even to map
+Knative and external resources under the same domain name.
+
+In this sample, we set up two web services: _Search_ service and _Login_
+service, which simply read in an env variable `SERVICE_NAME` and print
+`"${SERVICE_NAME} is called"`. We'll then define routing rules via [Kong's
+Kubernetes Ingress
+Controller](https://docs.konghq.com/kubernetes-ingress-controller/) and [Kong's
+request transformer
+plugin](https://docs.konghq.com/hub/kong-inc/request-transformer/), so that
+`example.com/search` maps to the _Search_ service, and `example.com/login` maps
+to the _Login_ service.
```
```suggestion
# Routing across multiple Knative Services with Kong

This example shows how to map multiple Knative Services to different paths under
a single domain name using [Kong Gateway](https://docs.konghq.com/gateway-oss/).
Kong Gateway is a general-purpose reverse proxy and API gateway built on
[Nginx](https://www.nginx.com/), therefore these directions can also be used to
configure routing based on other request data such as headers, or even to map
Knative and external resources under the same domain name.

This sample sets up two web services: _Search_ service and _Login_
service, which read an environment variable called `SERVICE_NAME`, and print
`"${SERVICE_NAME} is called"`. The sample then defines routing rules by using [Kong's
Kubernetes Ingress
Controller](https://docs.konghq.com/kubernetes-ingress-controller/) and [Kong's
request transformer
plugin](https://docs.konghq.com/hub/kong-inc/request-transformer/), so that
`example.com/search` maps to the _Search_ service, and `example.com/login` maps
to the _Login_ service.
```

abrennan89 commented:
/ok-to-test

abrennan89 commented:
TODO: Awaiting SME/technical review. Will follow up with an issue for docs writers to clean up formatting etc.

sisp commented:
I've addressed @abrennan89's comment and also fixed some bad Markdown formatting that caused wrong indentation in the rendered HTML.

Let me know if there is anything else you'd like done differently.

dprotaso commented:
/lgtm
/approve

dprotaso commented:
/retest

sisp commented:
Tests appear to be failing because of a timeout related to:
```bash
gem install bundler && bundle install
```

sisp commented:
/retest

dprotaso commented:
This landed so it should unblock us https://github.com/knative/test-infra/pull/2889

/retest
